workflows/ampliseq.nf

```diff
@@ -367,12 +383,32 @@ workflow AMPLISEQ {
     }
 
     //
-    // SUBWORKFLOW / MODULES : Taxonomic classification with DADA2 and/or QIIME2
+    // Modules : ITSx - cut out ITS region if long ITS reads
+    //
+    ch_full_fasta = ch_dada2_fasta
+    if (params.cut_its == "none") {
+        ch_fasta = ch_dada2_fasta
+    } else {
+        if (params.cut_its == "full") {
+            outfile = params.its_partial ? "ASV_ITS_seqs.full_and_partial.fasta" : "ASV_ITS_seqs.full.fasta"
+        }
+        else if (params.cut_its == "its1") {
+            outfile =  params.its_partial ? "ASV_ITS_seqs.ITS1.full_and_partial.fasta" : "ASV_ITS_seqs.ITS1.fasta"
+        }
+        else if (params.cut_its == "its2") {
+            outfile =  params.its_partial ? "ASV_ITS_seqs.ITS2.full_and_partial.fasta" : "ASV_ITS_seqs.ITS2.fasta"
+        }
+        ITSX_CUTASV ( ch_full_fasta, outfile )
+        ch_versions = ch_versions.mix(ITSX_CUTASV.out.versions.ifEmpty(null))
+        ch_fasta = ITSX_CUTASV.out.fasta
+    }
+
+    //
+    // SUBWORKFLOW / MODULES : Taxonomic classification with DADA2, SINTAX and/or QIIME2
```

Comment:
Well, that whole following section gets too complicated for my taste. I do not feel confident to review that at all.
This needs to be refactored to one or even up to 4 (1 per method: DADA2, Sintax, QIIME2, Phyloplacement?) subworkflow I think. This will make it much easier to read I hope.
Do you want to attempt this? Otherwise I might need to try that after that PR...

Reply:
I fully agree that it too complicated now, but wasn't sure what to do about it. I'll make an attempt with subworkflows.

Reply:
OK, I've added subworkflows, which makes ampliseq.nf a bit more readable I hope. I chose not to add a qiime2-taxonomy workflow, as that part already consists of two subworkflows and is quite clean. However, now the complicated things are in the DADA2-taxonomy subworkflow instead. I tried adding more comments, but I'm afraid it's still kind of convoluted...